### PR TITLE
Dynamically add java handlers to velocity template

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag [hunk 1]
```diff
@@ -26,7 +28,7 @@ require('/modules/store.js').exec(function (ctx) {
             options = {},
             types = store.assetTypes(),
             search = require('/modules/search.js'),
-            utility = require('/modules/util.js');
+            utility = require('/modules/util.js'),isSubscribed;
```
Comment (rushmin) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: Move isSubscribed variable declaration to the next line, for the sake of readability.

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag [hunk 2]
```diff
@@ -41,6 +43,121 @@ require('/modules/store.js').exec(function (ctx) {
...
+    }
+
+    var getEnterprises = function (asset) {
+        var store = jagg.module("manager").getAPIStoreObj();
+        var isEmailLoginEnabled = store.isEnableEmailUsername();
+
+        var appName = asset.attributes.overview_name;
+        var appVersion = asset.attributes.overview_version;
+        var appProvider = asset.attributes.overview_provider;
+        var result = null;
```
Comment (rushmin) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: 'result' is not a meaningful name.

Comment (lahiru823) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: this code has been removed in the latest code.

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag (hunk 1 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: commit# https://github.com/wso2/carbon-appmgt/commit/ae1afb1d9c0d4a5b272cbca6ad5e3263c1a4f71a

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag [hunk 3]
```diff
@@ -41,6 +43,121 @@ require('/modules/store.js').exec(function (ctx) {
     paging = store.assetsPaging(request);
     query = query ? search.build(String(query)) : null;
 
+
```
Comment (dinusha-dilrukshi) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: Remove unwanted line spacing.

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag [hunk 4]
```diff
@@ -41,6 +43,121 @@ require('/modules/store.js').exec(function (ctx) {
...
+        // Get the enterprise subscription the individual subscription for this user for this app.
+        var appInfo = {};
+        appInfo["name"] = asset.attributes.overview_name;
+        appInfo["version"] = asset.attributes.overview_version;
+        appInfo["provider"] = asset.attributes.overview_provider;
+
+        var defaultApplicationName = "DefaultApplication";
+
+        var enterpriseSubscription;
+        if (authenticator.getLoggedInUser()) {
```
Comment (dinusha-dilrukshi) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: Re-check the logic here. Same if condition has repeated twice.

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag [hunk 5]
```diff
@@ -41,6 +43,121 @@ require('/modules/store.js').exec(function (ctx) {
...
+        var ssoProviderInfo = asset.attributes.sso_ssoProvider.split("-");
+
+
+        var authResult = subscriptionService.getEnterprisesForApplication(
+                {appName: issuerName, ssoProviderName: ssoProviderInfo[0], ssoProviderVersion: ssoProviderInfo[1]});
+
+        return authResult.enterprises;
+
+    }
+
```
Comment (dinusha-dilrukshi) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: Remove unwanted line spacing.

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag (hunk 3 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: Already fixed in latest code

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag (hunk 5 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: This section of code is been removed in the latest code

File: modules/jaggery-apps/app-store-web/controllers/top-assets.jag (hunk 4 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-store-web/controllers/top-assets.jag]: Fixed with commit# https://github.com/wso2/carbon-appmgt/commit/797b30cf9afba3fd570826f199aa72722bd790ed

**Pull request merged**